connector.go

```diff
@@ -36,11 +36,18 @@ func (c *connector) Connect(ctx context.Context) (driver.Conn, error) {
 	dialsLock.RLock()
 	dial, ok := dials[mc.cfg.Net]
 	dialsLock.RUnlock()
+
 	if ok {
-		mc.netConn, err = dial(ctx, mc.cfg.Addr)
+		dctx := ctx
+		if mc.cfg.Timeout > 0 {
+			var cancel context.CancelFunc
+			dctx, cancel = context.WithTimeout(ctx, c.cfg.Timeout)
+			defer cancel()
+		}
+		mc.netConn, err = dial(dctx, mc.cfg.Addr)
 	} else {
 		nd := net.Dialer{Timeout: mc.cfg.Timeout}
-		mc.netConn, err = nd.DialContext(ctx, mc.cfg.Net, mc.cfg.Addr)
+		mc.netConn, err = nd.DialContext(ctx, mc.cfg.Net, mc.cfg.Addr) // for DialContext shall be ctx, not dialCtx
```

Comment:
This comment is redundant.

Reply:
yes, this explains why the dialctx not passed to DialContext, for later puller could get some info.
if you can't accept this redundant comment, I would comit another pach to remove this comment.

remove ?

Reply:
> this explains why the dialctx not passed to DialContext, for later puller could get some info.

Read the comment.  It doesn't explain *why*.  And the code tells *why* cleary.
So this comment is just a noise.  Please remove it.

Reply:
updated